[2.10] Misc other updates to guides (#4032)
astronomer/docs: learn/debugging-dags.md

- Contains either the word `airflow` or the word `dag`. The scheduler only parses files fulfilling this condition.
- Is called when defined with the `@dag` decorator. See also [Introduction to Airflow decorators](airflow-decorators.md).

> Note that in Airflow 2.10+ it is possible to configure an Airflow listeners as a plugin to run any Python code, either when a new import error appears (`on_new_dag_import_error`) or when the dag processor finds a known import error (`on_existing_dag_import_error`). See [Airflow listeners](using-airflow-plugins.md#airflow-listeners) for more information on Airflow listeners.

Reviewer: ```suggestion
Note that in Airflow 2.10+, you can configure an Airflow listener as a plugin to run any Python code, either when a new import error appears (`on_new_dag_import_error`) or when the dag processor finds a known import error (`on_existing_dag_import_error`). See [Airflow listeners](using-airflow-plugins.md#airflow-listeners) for more information.
```

Reviewer: Note that Michael is also updating this one, so you may have a merge conflict
